### PR TITLE
Bugfix/patro23 43 fix validation in import/export budgets

### PR DESCRIPTION
**Pull request by LukaszWinowicz**
https://tracker.intive.com/jira/browse/PATRO23-390
https://tracker.intive.com/jira/browse/PATRO23-395

### CODE REVIEW

**Review by IzabelaPalubicka — APPROVED** [review]

**Review by pzajaczkowski — APPROVED** [review]

File: src/modules/budget/application/Budget/ExportingBudgets/ExportResult.cs
Comment (pzajaczkowski) [inline src/modules/budget/application/Budget/ExportingBudgets/ExportResult.cs:1]: Move to shared in budgets please, will be used in exporting transactions aswell

**Review by pzajaczkowski — APPROVED** [review]

File: src/modules/budget/application/Budget/Shared/Services/BudgetImportService.cs [hunk 1]
```diff
@@ -62,20 +62,20 @@ public async Task<GetImportResult> Import(IFormFile file)
...
 
-		var download = await this.blobStorageService.DownloadFromBlobStorage(fileName);
-		using var reader = new StreamReader(download.Content);
-		using var csvReader = new CsvReader(reader, csvConfig);
-		await csvReader.ReadAsync();
-		var budgetsToImport = csvReader.GetRecords<GetBudgetTransferInfo>();
-		var budgetsAggregateList = await this.budgetDataService.ConvertBudgetsFromCsvToBudgetAggregate(budgetsToImport, csvConfig);
+		var budgetsAggregateList = await this.budgetDataService.ConvertBudgetsToBudgetAggregate(budgetInfos.Result.BudgetsList);
+
+		if (budgetInfos.Result.BudgetsErrorsList.Count == 0)
```
Comment (IzabelaPalubicka) [inline src/modules/budget/application/Budget/Shared/Services/BudgetImportService.cs:75]: if(!budgetInfos.Result.BudgetsErrorsList.Any())
```suggestion
		if (budgetInfos.Result.BudgetsErrorsList.Count == 0)
```

**Review by IzabelaPalubicka — APPROVED** [review]

File: src/modules/budget/api/Controllers/BudgetController.cs [hunk 2]
```diff
@@ -540,11 +540,16 @@ public async Task<IActionResult> ImportBudgets(IFormFile file)
 		var getImportResult = await this.budgetImportService.Import(file);
 		await this.commandBus.Send(getImportResult.BudgetAggregateList);
 
-		if (getImportResult.ImportResult.Uri != "No budgets were saved.")
+		if (getImportResult.ImportResult.Uri == "No budgets were saved.")
+		{
+			return this.BadRequest(new { Errors = getImportResult.ImportResult.ErrorsList, getImportResult.ImportResult.Uri });
+		}
+
+		if (getImportResult.ImportResult.Uri == "All budgets were saved.")
```
[the remark this answers is not in the document]
Comment (mateuszgiedzinski) [reply src/modules/budget/api/Controllers/BudgetController.cs:548]: Under Shared.Infrastructure we could have Exporting and Importing namespaces to share that across whole solution.

File: src/modules/budget/api/Controllers/BudgetController.cs [hunk 3]
```diff
@@ -508,7 +508,7 @@ public async Task<IActionResult> AddUsersToBudget([FromRoute] Guid budgetId, [Fr
 	/// <returns>A string containing the URI to Azure Blob Storage of the exported file.</returns>
```
Comment (mateuszgiedzinski) [inline src/modules/budget/api/Controllers/BudgetController.cs:508]: First of all it should be fixed under new bug not closed story.

File: src/modules/budget/application/Budget/Shared/Services/BudgetDataService.cs [hunk 4]
```diff
@@ -20,29 +19,28 @@ namespace Intive.Patronage2023.Modules.Budget.Application.Budget.Shared.Services
...
+	private readonly ICsvService<GetBudgetTransferInfo> csvService;
 
 	/// <summary>
 	/// Initializes a new instance of the <see cref="BudgetDataService"/> class.
 	/// </summary>
 	/// <param name="contextAccessor">Provides access to the execution context, allowing the service to use user-specific information.</param>
-	/// <param name="blobStorageService">Provides functionality for interacting with Azure Blob Storage.</param>
 	/// <param name="queryBus">Enables the service to dispatch queries to the appropriate handlers.</param>
-	public BudgetDataService(IExecutionContextAccessor contextAccessor, IBlobStorageService blobStorageService, IQueryBus queryBus)
+	/// <param name="csvService">ICsvService.</param>
```
Comment (mateuszgiedzinski) [inline src/modules/budget/application/Budget/Shared/Services/BudgetDataService.cs:30]: Never write that param documentation.  You can always find something better than string with parameter type.

File: src/modules/budget/application/Budget/Shared/Services/BudgetDataService.cs [hunk 5]
```diff
@@ -65,7 +63,7 @@ public Task<BudgetAggregateList> ConvertBudgetsFromCsvToBudgetAggregate(IEnumera
 	/// If a budget with the same name already exists in the database, a random number is appended to the name.
 	/// </summary>
 	/// <param name="budget">The budget information used to create the new budget.</param>
-	/// <param name="budgetsNames">The budget information used to create the new budget2.</param>
+	/// <param name="budgetsNames">The existing budgets' names used for checking whether the new budget's name already exists in the database.</param>
```
Comment (mateuszgiedzinski) [inline src/modules/budget/application/Budget/Shared/Services/BudgetDataService.cs:66]: typo budgets' => budget's

File: src/modules/budget/application/Budget/Shared/Services/IBudgetDataService.cs [hunk 6]
```diff
@@ -13,16 +13,15 @@ public interface IBudgetDataService
...
-	Task<BudgetAggregateList> ConvertBudgetsFromCsvToBudgetAggregate(IEnumerable<GetBudgetTransferInfo> budgetsToImport, CsvConfiguration csvConfig);
+	Task<BudgetAggregateList> ConvertBudgetsToBudgetAggregate(IEnumerable<GetBudgetTransferInfo> budgetsToImport);
 
 	/// <summary>
 	/// Creates a new budget based on the provided budget information.
 	/// If a budget with the same name already exists in the database, a random number is appended to the name.
 	/// </summary>
 	/// <param name="budget">The budget information used to create the new budget.</param>
-	/// <param name="budgetsNames">The budget information used to create the new budget2.</param>
+	/// <param name="budgetsNames">The existing budgets' names used for checking whether the new budget's name already exists in the database.</param>
```
Comment (mateuszgiedzinski) [inline src/modules/budget/application/Budget/Shared/Services/IBudgetDataService.cs:24]: typo

File: src/modules/budget/application/Budget/Shared/Services/IBudgetDataService.cs [hunk 7]
```diff
@@ -13,16 +13,15 @@ public interface IBudgetDataService
 	/// Converts a collection of budget information from CSV format into a list of BudgetAggregate objects.
 	/// </summary>
 	/// <param name="budgetsToImport">Collection of budget information to be converted, represented as GetBudgetTransferInfo objects.</param>
-	/// <param name="csvConfig">Configuration for reading the CSV file.</param>
 	/// <returns>A Task containing a BudgetAggregateList, representing the converted budget information.</returns>
-	Task<BudgetAggregateList> ConvertBudgetsFromCsvToBudgetAggregate(IEnumerable<GetBudgetTransferInfo> budgetsToImport, CsvConfiguration csvConfig);
+	Task<BudgetAggregateList> ConvertBudgetsToBudgetAggregate(IEnumerable<GetBudgetTransferInfo> budgetsToImport);
```
Comment (mateuszgiedzinski) [inline src/modules/budget/application/Budget/Shared/Services/IBudgetDataService.cs:17]: MapFrom

File: src/modules/budget/application/Budget/ExportingBudgets/ExportResult.cs
Comment (mateuszgiedzinski) [reply src/modules/budget/application/Budget/ExportingBudgets/ExportResult.cs:1]: Under Shared.Infrastructure we could have Exporting and Importing namespaces to share that across whole solution.

**Review by mateuszgiedzinski — CHANGES_REQUESTED** [review]

File: src/modules/budget/api/Controllers/BudgetController.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (LukaszWinowicz) [reply src/modules/budget/api/Controllers/BudgetController.cs:548]: You have right, I removed this.

**Review by mateuszgiedzinski — APPROVED** [review]

**Pull request merged**